### DISCUSSION TITLE
Genesis save read not working and inconsistent

### DISCUSSION THREAD

**Discussion by kdawg8** [sanni/cartreader#236] [Need help]
I have a few cart readers and the saves they are dumping from Sonic the Hedgehog 3 are not valid (i.e. does not work in an emulator with or without the 8to16 conversion suggested in the wiki) and inconsistent between dumps.  They both dumps ROMs and saves fine for everything else, and the genesis ROM hashes match the DoM database so I don't think there is a hardware issue (unless there are pins used for save reads but not ROM reads?).  I also have a retrode, and the save which is dumped from this game on that works in an emulator.  Comparing the files from the two pieces of hardware they are very different.  The cart reader files contain 0x25B or 0x1FF bytes (length is consistent between dumps but different between the two cart readers) while the retrode file contains 0x3FF bytes, the first difference between the retrode and cart reader save files is within the first 16 bytes and I don't see anything systematic about the differences between the files.  Please help, thanks!

**Comment by sanni** [comment]
I did test Sonic 3 in 2017. Sadly I don't have the game anymore to check if it still works with the latest version of the code. Sonic 3 is a special case since the SRAM needs to be activated first. It could be that this function did break sometime between 2017 and now.
Maybe someone else with Sonic 3 can test their cartridge.

**Reply by kdawg8** [reply]
Thanks sanni.  I'm going to sell my extra cart readers and Sonic 3 is the only genesis game I own with battery backup I can use for testing, didn't realize it was a special case.  I might check a local game store to see if they have a different genesis game with battery backup to see if its just Sonic 3 with the problem...can't advertise it can backup saves if I know this isn't working.

If the function has indeed broken over time, do you have any suspicion if its hardware or software?  I could try loading up old versions of of the cart reader software and see if it starts working.

**Reply by kdawg8** [reply]
Update: I bought another Genesis cart with battery backup (Starflight) and the cart readers are able to backup the save and it works in an emulator.
